### PR TITLE
[NEW] Header Redesign

### LINKED ISSUE #7772: rocket.chat doesnt load on mobile
* Version of Rocket.Chat Server: 0.58.0
* Operating System: ubuntu
* Deployment Method(snap/docker/tar/etc):
* Number of Running Instances:  1 
* DB Replicaset Oplog: yes
* Node Version: 4.8.3

rocket .chat doesnt load on mobile isnt responsive 

telechargement

### LINKED ISSUE #7776: Long channel topic leads to over-wide window
Just updated to 0.58.0, the channels with longer topics show following behaviour:

Top bar (#channelname topic...) resizes to fit topic (single row). The message-UI also resizes to match top bar. Both run out of the browser window and the overflowing content is hidden; you can't scroll horizontally, other problem is, that you can only change the topic from the Rocket.Chat App, as the sidebar disappears too.
Somehow - if you managed it to change the topic to something shorter - the items in the sidebar are no longer clickable on a desktop client. [Separate issue?]
Screenshots are in referenced issue #7772

### LINKED ISSUE #8075: Long topic names causes sidebar and send button to disappear on Firefox
Right sidebar and send button have gone off-page. Only happens on Firefox, not Chrome. 
0.59.0-develop, e568f8a

### PR DESCRIPTION
**Pull request by MartinSchoeler**
@RocketChat/core 

Parts done so far: 
- [x] Direct message title
screen shot 2017-08-15 at 3 48 27 pm

- [x] Channel Title
screen shot 2017-09-13 at 10 41 09 am

- [x] Administration and account headers

screen shot 2017-09-13 at 10 41 26 am

<details>
 <summary>More Screenshots...</summary>
screen shot 2017-09-13 at 10 41 51 am

screen shot 2017-09-13 at 10 46 54 am

screen shot 2017-09-13 at 10 47 12 am

screen shot 2017-09-13 at 10 47 25 am

</details>
---

closes #7776 
closes #7772
closes #8075

### CODE REVIEW

File: packages/rocketchat-theme/client/imports/components/header.css [hunk 1]
```diff
@@ -0,0 +1,159 @@
+.rc-header {
+	display: flex;
+
+	height: 77px;
+	padding: 20px;
+
+	color: var(--color-dark);
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/components/header.css]: Why didn't you create a variable for this?

File: packages/rocketchat-theme/client/imports/components/header.css [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+			&--offline {
+				background-color: var(--status-invisible);
+			}
+		}
+	}
+
+	&__toggle-favorite {
+		color: var(--header-toggle-favorite-star-color);
+		&.--empty {
```
Comment (karlprieb) [inline packages/rocketchat-theme/client/imports/components/header.css]: I think this cl;ass don't exist.

File: packages/rocketchat-ui/client/views/app/room.js [hunk 3]
```diff
@@ -50,6 +50,14 @@ const openProfileTabOrOpenDM = (e, instance, username) => {
 };
 
 Template.room.helpers({
+	avatarUrl() {
```
Comment (karlprieb) [inline packages/rocketchat-ui/client/views/app/room.js]: You're returning an avatar background and not the avatar url, change the name or return only the url :)

**Comment by geekgonecrazy** [comment]
@rocketchat/core is this going to make it in 0.59.0 ? If not DM's won't show status which is a Breaking change

**Comment by sampaiodiego** [comment]
right @geekgonecrazy .. I think this **have** to make to be `0.59`

File: packages/rocketchat-theme/client/imports/components/header.css [hunk 4]
```diff
@@ -0,0 +1,214 @@
+.rc-header {
+	display: flex;
+
+	max-width: 100vw;
+	height: 77px;
```
Comment (ggazzo) [inline packages/rocketchat-theme/client/imports/components/header.css]: I think we should use a variable here

File: packages/rocketchat-theme/client/imports/components/header.css [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+	padding: 20px 24px;
```
Comment (ggazzo) [inline packages/rocketchat-theme/client/imports/components/header.css]: here to.

File: packages/rocketchat-theme/client/imports/components/header.css [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+	&__section--button {
+		display: flex;
+		flex: 1 1 100%;
+		justify-content: flex-end;
+
+		/* max-width: use this to allign the buttons with the form */
+	}
+
+	&__section--button > * {
```
Comment (ggazzo) [inline packages/rocketchat-theme/client/imports/components/header.css]: na na na, avoid `*` selectors =/

Comment (ggazzo) [reply packages/rocketchat-theme/client/imports/components/header.css]: avoid not, dont use :P

File: packages/rocketchat-theme/client/imports/components/header.css [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+	display: none;
+	visibility: hidden;
+
+	margin-right: 20px;
+
+	cursor: pointer;
+	transition: transform 0.2s ease-out 0.1s;
+	will-change: transform;
+
+	& i {
```
Comment (ggazzo) [inline packages/rocketchat-theme/client/imports/components/header.css]: use a class instead tag name.

File: packages/rocketchat-theme/client/imports/components/header.css [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+		text-align: center;
+
+		border-radius: 20px;
+
+		font-size: 12px;
+		font-weight: bold;
+		line-height: 18px;
+	}
+
+	&.menu-opened i {
```
Comment (ggazzo) [inline packages/rocketchat-theme/client/imports/components/header.css]: use a class instead tag name.

File: packages/rocketchat-ui-admin/client/admin.js [hunk 9]
```diff
@@ -355,7 +355,7 @@ Template.admin.events({
 		const value = _.trim($(e.target).val());
 		TempSettings.update({ _id: this._id }, { $set: { editor: value }});
 	},
-	'click .submit .discard'() {
+	'click .rc-header__section--button .discard'() {
```
Comment (ggazzo) [inline packages/rocketchat-ui-admin/client/admin.js]: just to remember to talk about ;)

File: packages/rocketchat-ui/client/components/header/header.html [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+				{{/if}}
+			{{/if}}
+
+			{{#if isSection}}
+				<span class="rc-header__section--title">{{_ sectionName}}</span>
+			{{/if}}
+		</div>
+
+		{{#if buttons}}
+			{{> Template.contentBlock}}
```
Comment (ggazzo) [inline packages/rocketchat-ui/client/components/header/header.html]: amazing

File: packages/rocketchat-ui/client/components/header/header.js [hunk 11]
```diff
@@ -0,0 +1,124 @@
+/* globals fireGlobalEvent*/
```
Comment (ggazzo) [inline packages/rocketchat-ui/client/components/header/header.js]: maybe we must rename this file to RoomHeader

Comment (MartinSchoeler) [reply packages/rocketchat-ui/client/components/header/header.js]: @ggazzo  But this header can be a section header too, like the ones in the preferences and settings screens

**Pull request merged**